api/v1alpha4/condition_consts.go

```diff
@@ -184,3 +184,38 @@ const (
 	// from making any further remediations.
 	TooManyUnhealthyReason = "TooManyUnhealthy"
 )
+
+// These are valid conditions of a MachineDeployment.
+const (
+	// Available means the MachineDeployment is available, ie. at least the minimum available
+	// machines required are up and running for at least minReadySeconds.
+	MachineDeploymentAvailable ConditionType = "MachineDeploymentAvailable"
+	// Progressing means the MachineDeployment is progressing. Progress for a MachineDeployment is
+	// considered when a new machine set is created or adopted, and when new machine scale
+	// up or old machine scale down. Progress is not estimated for paused deployments or
+	// when progressDeadlineSeconds is not specified.
+	MachineDeploymentProgressing ConditionType = "MachineDeploymentProgressing"
+	// MachineFailure is added in a MachineDeployment when one of its machines fails to be created
+	// or deleted.
+	MachineDeploymentMachineFailure ConditionType = "MachineDeploymentMachineFailure"
```

Comment:
These two conditions seem a bit out of place compared to others, given that they could be a reason for `Ready` or similar, rather than on their own. @fabriziopandini what do you think?

Reply:
Just to give some context. 
The code is modeled after the [Deployment/ReplicaSet code](https://github.com/kubernetes/kubernetes/blob/fa2787acb5f5bba6981556c733c38c8968c339d2/staging/src/k8s.io/api/apps/v1/types.go#L426-L429)

Additionally, the [much like ](https://github.com/kubernetes/kubectl/blob/c1df07341a15fb36264a018e130ee397332bdfc0/pkg/polymorphichelpers/rollout_status.go#L76)the `kubectl rollout status` command, the `clusterctl rollout status` will monitor the Progressing condition.

Reply:
Got it, although to be consistent with the rest of CAPI conditions, we might want to change things up a little bit. I'd see having a `Ready` condition on MachineDeployment that not only aggregates the status of each Machine in a MachineDeployment, but can tell users if there is a failure or a scaling operation is in progress. This would be a `Reason` for the condition to be `False`

Reply:
I agree with Vince we should aim to better consistency, and in this case, we should revisit things in order to have all the conditions using a positive polarity.

Please note there is also the option to propose an amendment to the current condition proposal, so we can discuss pros and cons and address all the related concerns e.g. how do we compute summary in case of conditions with mixed polarity? what are the required changes to util/conditions in order to help users in shifting to the new model?

Reply:
+1 on ensuring we have the `Ready` condition. This is for consistency and making this kstatus compatible.

Reply:
If `Progressing` a way to describe that a rolling upgrade is happening, should we be more explicit here? Thinking about the rest of the system as well, should we do something similar in KCP and express better when a resource is performing a rolling operation? 

Before merging and proceeding with any of these, I'd like to find some consistency and apply it everywhere.